pkg/mqtrigger/mqtrigger.go

```diff
@@ -40,10 +45,57 @@ func Start(logger *zap.Logger, routerUrl string) error {
 	// Message queue type: nats is the only supported one for now
 	mqType := os.Getenv("MESSAGE_QUEUE_TYPE")
 	mqUrl := os.Getenv("MESSAGE_QUEUE_URL")
+
+	// For authentication with message queue
+	secrets, err := readSecrets(logger)
+	if err != nil {
+		return err
+	}
+
 	mqCfg := messageQueue.MessageQueueConfig{
-		MQType: mqType,
-		Url:    mqUrl,
+		MQType:  mqType,
+		Url:     mqUrl,
+		Secrets: secrets,
 	}
 	messageQueue.MakeMessageQueueTriggerManager(logger, fissionClient, routerUrl, mqCfg)
 	return nil
 }
+
+func readSecrets(logger *zap.Logger) (map[string][]byte, error) {
+	secretsPath := strings.TrimSpace(os.Getenv("MESSAGE_QUEUE_SECRETS"))
+	// return if no secretsPath is present
+	if len(secretsPath) == 0 {
+		return nil, nil
+	}
+
+	// return if no secrets exist
+	if _, err := os.Stat(secretsPath); os.IsNotExist(err) {
+		return nil, nil
```

Comment:
We should only run the function when `TLS_ENABLED` is true, so return the error here.
In this way, we can distinguish the situations if it's `TLS_ENABLED=false` or user gave the wrong path to volume.

Reply:
If we want to use `readSecrets` as a generic function (i.e., use them for other messaging queues as well), it would make more sense to log that the secretsPath was not found and let it return nil. 

There are two cases which are of importance here:
1. When no secrets are present and they are not required. i.e., there is no need for secrets e.g. connecting to a non TLS secured Kafka cluster would not need secrets.
2. When no secrets are present but they should be present (secrets are required). This would be the case when the Kafka cluster is to be connected using TLS but the secrets were not mounted properly.

Returning nil should handle the first scenario just fine because there will be no error and throwing an error is not really required here. Problem arises in the second case. If the secrets cannot be found at the path but they should be present there, an error should be raised so that we know what went wrong. Returning nil here would be just ignoring this case and it would make it difficult to debug the problem. 

I believe returning error is what will handle the second case. But returning error when secrets will not be required (connecting to non TLS Kafka cluster) and distinguishing this harmless thing with an actual error would be difficult. 

One way to fix this can be, we log the information (maybe as a warning) so that we know that the secrets were not found and hope that the messaging queue would throw an error which together with the warning would help debug the problem. We can word the warning to convey the intensity of the problem. I am not very happy with this fix but this is something which can solve the problem for now. 

What do you think?

Reply:
I don't quite get it. If the secrets are not necessary, why calling this function?

It's the caller's responsibility to decide whether it needs to call the function, instead of calling a function that will silently ignore the error and pretend nothing happen. A generic function should always return the error instead of hiding it (unless the function itself can handle it) and let the caller to decide what to do to handle the error.

In this case, I think we can change the function signature

```
// caller checks whether it need to read secrets
secretsPath := strings.TrimSpace(os.Getenv("MESSAGE_QUEUE_SECRETS"))
if secretsPath > 0 {
    secrets, err := readSecrets(secretsPath, logger) (map[string][]byte, error)
}


// change the function signature to
func readSecrets(path string, logger *zap.Logger) (map[string][]byte, error) {

    // return if no secrets exist
	if _, err := os.Stat(path); os.IsNotExist(err) {
        return nil, err
    }
....
```

It brings a couple of benefits,

1. A more generic function that can accept any path instead of reading from the environment variable.
2. It's easier to write unit tests for the function.

Reply:
This makes sense. :+1: